src/pipecat/services/elevenlabs/tts.py

```diff
@@ -254,13 +254,15 @@ def _set_voice_settings(self):
     async def set_model(self, model: str):
         await super().set_model(model)
         logger.info(f"Switching TTS model to: [{model}]")
-        # No need to disconnect/reconnect for model changes with multi-context API
+        await self._disconnect()
+        await self._connect()
 
     async def _update_settings(self, settings: Mapping[str, Any]):
         prev_voice = self._voice_id
         await super()._update_settings(settings)
-        # If voice changes, we don't need to reconnect, just use a new context
         if not prev_voice == self._voice_id:
+            await self._disconnect()
```

Comment:
We should move this after the log, so it's easier to know what cause the disconnection.